lib/statsig/configs.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Statsig.Configs do
+  use GenServer
+  require Logger
+
+  @table_name :statsig_configs
+
+  @type state :: %{
+    last_sync_time: integer(),
+    reload_timer: reference() | nil
+  }
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, %{
+      last_sync_time: 0,
+      reload_timer: nil,
+    }, name: __MODULE__)
+  end
+
+  @impl true
+  def init(state) do
+    Logger.error("Initializing Statsig.Configs #{inspect(Application.get_all_env(:statsig))}")
+    Logger.error("Stacktrace: #{inspect(Process.info(self(), :current_stacktrace))}")
+    :ets.new(@table_name, [:named_table, :set, :public])
+    timer = state.reload_timer || Process.send_after(self(), :reload_configs, get_reload_interval())
+    new_state = Map.put(state, :reload_timer, timer)
+
+    case reload_configs(new_state) do
+      {:ok, updated_state} -> {:ok, updated_state}
+      {:error, :reload_failed, _error} -> {:ok, state}
+      {:error, reason} -> {:ok, state}
+    end
+  end
+
+  @impl true
+  def handle_info(:reload_configs, state) do
+    timer = Process.send_after(self(), :reload_configs, get_reload_interval())
+    new_state = Map.put(state, :reload_timer, timer)
+
+    case reload_configs(new_state) do
+      {:ok, updated_state} -> {:noreply, updated_state}
+      {:error, :reload_failed, _error} -> {:noreply, state}
```

Comment:
any reason the reload timer is dropped here?
This is why i strongly recommend pulling apart messaging and state logic.